sampledb/logic/schemas/validate.py

```diff
@@ -310,36 +310,67 @@ def _validate_quantity(instance: dict, schema: dict, path: typing.List[str]) ->
     if not units_are_valid(instance['units']):
         raise ValidationError('Invalid/Unknown units', path)
-    if not isinstance(instance['magnitude_in_base_units'], float) and not isinstance(instance['magnitude_in_base_units'], int):
-        raise ValidationError('magnitude_in_base_units must be float or int', path)
-    try:
-        quantity = datatypes.Quantity(instance['magnitude_in_base_units'], units=instance['units'])
-    except Exception:
-        raise ValidationError('Unable to create quantity', path)
-    if not isinstance(instance['dimensionality'], str):
-        raise ValidationError('dimensionality must be str', path)
     try:
         schema_quantity = datatypes.Quantity(1.0, units=schema['units'])
     except Exception:
         raise ValidationError('Unable to create schema quantity', path)
-    if quantity.dimensionality != schema_quantity.dimensionality:
+
+    quantity_magnitude = None
+    quantity_magnitude_in_base_units = None
+    if 'magnitude' in instance:
+        if not isinstance(instance['magnitude'], float) and not isinstance(instance['magnitude'], int):
+            raise ValidationError('magnitude must be float or int', path)
+        try:
+            quantity_magnitude = datatypes.Quantity(instance['magnitude'], units=instance['units'],
+                                                    already_in_base_units=False)
+        except Exception:
+            raise ValidationError('Unable to create quantity based on given magnitude', path)
+    if 'magnitude_in_base_units' in instance:
+        if not isinstance(instance['magnitude_in_base_units'], float) and not isinstance(instance['magnitude_in_base_units'], int):
+            raise ValidationError('magnitude_in_base_units must be float or int', path)
+        try:
+            quantity_magnitude_in_base_units = datatypes.Quantity(instance['magnitude_in_base_units'], units=instance['units'],
+                                                                  already_in_base_units=True)
+        except Exception:
+            raise ValidationError('Unable to create quantity based on given magnitude_in_base_units', path)
+
+    if quantity_magnitude is not None and quantity_magnitude_in_base_units is not None \
+            and quantity_magnitude.magnitude != quantity_magnitude_in_base_units.magnitude:
+        raise ValidationError(
+            'magnitude and magnitude_in_base_units do not match, either set only one or make sure both match', None)
+    elif quantity_magnitude is None and quantity_magnitude_in_base_units is None:
+        raise ValidationError(
+            'missing keys in schema: either magnitude or magnitude_in_base_units has to be given', None)
+    elif quantity_magnitude is None:
+        quantity_magnitude = quantity_magnitude_in_base_units
+
+    # automatically add dimensionality and either magnitude oder magnitude_in_base_units, if they haven't been given yet
+    instance.update(quantity_magnitude.to_json())
```

Comment:
This will override `dimensionality` (and both magnitudes), even if it has been set in `instance`. Instead you could do something like:

```python
for key, value in quantity_magnitude.to_json().items():
    if key not in instance:
        instance[key] = value
```

which would only set values for keys missing in `instance`.

Reply:
Updated in the branch and pull request